tools/msg/nes/msgdisNES.py

```diff
@@ -0,0 +1,316 @@
+            0xFA: ("CMD_HS_DEKU_PLAYGROUND_DAY_1", self.decode_cmd_arg_none),
+            0xFB: ("CMD_HS_DEKU_PLAYGROUND_DAY_2", self.decode_cmd_arg_none),
+            0xFC: ("CMD_HS_DEKU_PLAYGROUND_DAY_3", self.decode_cmd_arg_none),
+            0xFD: ("CMD_DEKU_PLAYGROUND_NAME_DAY_1", self.decode_cmd_arg_none),
+            0xFE: ("CMD_DEKU_PLAYGROUND_NAME_DAY_2", self.decode_cmd_arg_none),
+            0xFF: ("CMD_DEKU_PLAYGROUND_NAME_DAY_3", self.decode_cmd_arg_none)
+        }
+        if self.decodedText != "":
+            return
+
+        prevText = False
+        prevNewline = True
+        prevCmd = False
+        self.decodePos = 0
+        textLen = len(self.text)
+        while self.decodePos < textLen:
+            char = self.text[self.decodePos]
+            self.decodePos += 1
+
+            if char >= 0x20 and char <= 0xBB: # Characters
+                if prevCmd:
+                    self.decodedText += ' "'
+                elif prevNewline:
+                    self.decodedText += '"'
+
+                if char == 0x22: # Handle escaping "
+                    self.decodedText += '\\"'
+                elif char >= 0xB0: # Button characters
+                    self.decodedText += f'{MessageNES.BUTTONMAP[char]}'
+                else:
+                    self.decodedText += chr(char)
+
+                prevText = True
+                prevNewline = False
+                prevCmd = False
+            elif char == 0x11: # New line
+                if prevCmd:
+                    self.decodedText += ' "'
+                elif prevNewline:
+                    self.decodedText += '"'
+
+                self.decodedText += f'\\n"\n'
+
+                prevText = False
+                prevNewline = True
+                prevCmd = False
+            elif char == 0x10 or char == 0x12 or char == 0x1B: # Box Breaks add automatic newlines
+                if prevText:
+                    self.decodedText += '"'
+
+                cmd, decoder = CMDMAP[char]
+                self.decodedText += f'\n{decoder(cmd)}\n'
+
+                prevText = False
+                prevNewline = True
+                prevCmd = False
+            elif char == 0xBF: # End command, do nothing
+                if prevText:
+                    self.decodedText += '"'
+            else: # Control Codes (see message_data_fmt_nes.h)
+                if prevText:
+                    self.decodedText += '" '
+                elif prevCmd:
+                    self.decodedText += ' '
+
+                cmd, decoder = CMDMAP[char]
+                self.decodedText += decoder(cmd)
+
+                prevText = False
+                prevNewline = False
+                prevCmd = True
+
+def parseTable(start):
+    table = {}
+    with open("baserom/code","rb") as f:
+        f.seek(start)
+        buf = f.read(8)
+        textId, typePos, segment = struct.unpack(">HBxI", buf)
+        while segment != 0:
```

Comment:
Reminder to self: Actually check if textId is` 0xFFFF` instead of segment is 0 like `Message_FindMessageNES`